packages/kotti-ui/source/kotti-comment/KtComment.vue

```diff
@@ -95,75 +104,74 @@ export default {
 		KtCommentInput,
 	},
 	props: makeProps(KottiComment.propsSchema),
-	data() {
-		return {
-			showInlineReply: false,
-			replyToText: null,
-			replyToUserId: Number,
-			isInlineEdit: false,
-			inlineMessageValue: '',
+	setup(props, { emit }) {
+		const isInlineEdit = ref(false)
+		const showInlineReply = ref(false)
+		const inlineMessageValue = ref<string | null>(null)
+		const userBeingRepliedTo = ref<UserData | null>(null)
+
+		const handleDelete = (commentId: number | string, isInline?: boolean) => {
+			emit('delete', {
+				id: commentId,
+				parentId: isInline ? props.id : null,
+			})
 		}
-	},
-	computed: {
-		styleObject() {
-			return {
-				avatar: true,
-				'avatar-selected': this.selected,
-			}
-		},
-		inlineMessage() {
-			return this.inlineMessageValue || this.message
-		},
-		actionOptions() {
-			const options = []
-			if (this.isEditable)
-				options.push({
-					label: 'Edit',
-					onClick: () => {
-						this.inlineMessageValue = this.inlineMessage
-						this.isInlineEdit = true
-					},
-					type: 'EDIT',
-				})
-			if (this.isDeletable)
-				options.push({
-					label: 'Delete',
-					onClick: () => this.handleDelete(this.id),
-					type: 'DELETE',
+		return {
+			actionOptions: computed<Array<KottiButton.Props>>(() =>
+				compact([
+					props.isEditable
+						? {
+								label: 'Edit',
+								onClick: () => {
+									if (inlineMessageValue.value === null)
+										inlineMessageValue.value = props.message
+									isInlineEdit.value = true
+								},
+								type: KottiButton.Type.PRIMARY,
+						  }
+						: null,
+					props.isDeletable
+						? {
+								label: 'Delete',
+								onClick: () => handleDelete(props.id),
+								type: KottiButton.Type.DANGER,
+						  }
+						: null,
+				]),
+			),
+			cancelInlineEdit: () => {
+				inlineMessageValue.value = null
+				isInlineEdit.value = false
+			},
+			handleDelete,
+			handleEditConfirm: () => {
+				isInlineEdit.value = false
+				if (!inlineMessageValue.value) return
```

Comment:
make `inlineMessageValue.value === null` please

we don't support `''` initialization anymore, so we can be explicit

Reply:
You are thinking one step ahead. `inlineMessageValue` is not bound by a `KtFieldTextArea` but by a `textarea` (which is designed to look like a `KtInlineEdit` lol). So the check is actually intentional since `''` is a possible value that might show up here.